include/gunrock/algorithms/hits.hxx

```diff
@@ -0,0 +1,375 @@
+                      this->auth_next.end(),
+                      thrust::square<float>());
+    this->sum =
+    thrust::reduce(policy,
+                   this->auth_next.begin(),
+                   this->auth_next.end());
+
+    op<float> op{this->sum};
+    thrust::for_each(policy,
+                     this->auth_next.begin(),
+                     this->auth_next.end(),
+                     op);
+  }
+
+  __device__
+  void norm_hub(){
+    auto policy = this->get_single_context(0)->execution_policy();
+
+    thrust::for_each(policy,
+                      this->hub_next.begin(),
+                      this->hub_next.end(),
+                      thrust::square<float>());
+    this->sum =
+    thrust::reduce(policy,
+                   this->hub_next.begin(),
+                   this->hub_next.end());
+
+    op<float> op{this->sum};
+    thrust::for_each(policy,
+                     this->hub_next.begin(),
+                     this->hub_next.end(),
+                     op);
+  }
+
+  thrust::device_vector<float>
+  get_auth(){
+    return this->auth_curr;
+  }
+
+  thrust::device_vector<float>
+  get_hub(){
+    return this->hub_curr;
+  }
+
+};// end of problem_c
+
+template <typename problem_t>
+struct enactor_t : gunrock::enactor_t<problem_t> {
+  using gunrock::enactor_t<problem_t>::enactor_t;
+
+  using vertex_t = typename problem_t::vertex_t;
+  using edge_t = typename problem_t::edge_t;
+  using weight_t = typename problem_t::weight_t;
+
+  void prepare_frontier(frontier_t<vertex_t>* f,
+                        cuda::multi_context_t& context) override {
+        //qqq need to prepare_frontier?
+  }
+
+  void loop(cuda::multi_context_t& context) override {
+    // Data slice qqq
+    auto E = this->get_enactor();
+    auto P = this->get_problem();
+    auto G = P->get_graph();
+
+    auto update = [P] __host__ __device__(
+                            vertex_t& source,
+                            vertex_t& neighbor,
+                            edge_t const& edge,
+                            weight_t const& weight
+                            ) ->bool{
+    P->update_hub(source, neighbor);
+    P->update_auth(neighbor, source);
+
+    return true;
+    };// end of update
+
+    // Execute advance operator on the provided lambda
+    operators::advance::execute<operators::load_balance_t::block_mapped,
+                                operators::advance_direction_t::forward,
+                                operators::advance_io_type_t::graph,
+                                operators::advance_io_type_t::vertices>(
+                    G, E, update, context);
+
+    // Normalize authority and hub
+    P->norm_auth();
+    P->norm_hub();
+
+    // Swap buffer
+    P->swap_buffer();
+
+    // Update iterator
+    P->update_iterator();
+
+  }// end of loop
+
+  bool is_converged(cuda::multi_context_t& context) override {
+    auto P = this->get_problem();
+    return P->is_converged();
```

Comment:
Why isn't the contents of `P->is_converged()` just part of this function instead? Why does it have to call a function within a function to do something?